### PR TITLE
Fix build warnings in NSOperation

### PR DESCRIPTION
**Pull request by pushkarnk**

### CODE REVIEW

File: Foundation/NSData.swift [hunk 1]
```diff
@@ -383,8 +383,8 @@ extension NSData {
         }
         
         
-        return other.withUnsafeBytes { (bytes2: UnsafePointer<Void>) -> Bool in
-            let bytes1 = bytes
```
Comment (pushkarnk) [inline Foundation/NSData.swift]: Though UnsafePointer<Void> needs to be replaced by UnsafeRawPointer, I don't think its applicable to ```withUnsafeBytes<ResultType, ContentType>(_ body: (UnsafePointer<ContentType>) throws -> ResultType)```

File: Foundation/NSOperation.swift [hunk 2]
```diff
@@ -326,9 +326,9 @@ open class OperationQueue: NSObject {
             } else {
                 effectiveName = "NSOperationQueue::\(Unmanaged.passUnretained(self).toOpaque())"
             }
-            let attr: DispatchQueueAttributes
+            let attr: DispatchQueue.Attributes
             if maxConcurrentOperationCount == 1 {
-                attr = .serial
+                attr = [] 
             } else {
```
Comment (pushkarnk) [inline Foundation/NSOperation.swift]: DispatchQueue.Attributes defines only ```.concurrent```

File: Foundation/NSOperation.swift [hunk 3]
```diff
@@ -408,7 +408,7 @@ open class OperationQueue: NSObject {
                 group.enter()
             }
 
-            let block = DispatchWorkItem(group: queueGroup, flags: .enforceQoS) { () -> Void in
+            let block = DispatchWorkItem(flags: .enforceQoS) { () -> Void in
                 if let sema = self._concurrencyGate {
```
Comment (pushkarnk) [inline Foundation/NSOperation.swift]: The new ```DispatchWorkItem(qos:flags:block:)``` does not need the queue group

**Comment by michael-lehew** [comment]
The Data parts of this are handled by: https://github.com/apple/swift-corelibs-foundation/pull/553 (which brings Data+NSData to mostly Swift3 parity).

**Comment by pushkarnk** [comment]
@michael-lehew : I have reverted the NSData changes from this PR. Thanks.

**Comment by phausler** [comment]
looks good to me; built manually on linux.

**Pull request merged**